### PR TITLE
Absolute urls should be handled by source maps

### PR DESCRIPTION
**Pull request by markfinger**
re: postcss#757
re: postcss#758

### CODE REVIEW

File: lib/map-generator.es6 [hunk 1]
```diff
@@ -165,6 +165,10 @@ export default class {
     }
 
     relative(file) {
+        if ( file.match(/https?:\/\/|file:\/\//) ) {
```
Comment (lydell) [inline lib/map-generator.es6]: `/^(?:https?|file):\/\//.test(file)`

Comment (lydell) [inline lib/map-generator.es6]: The most important point in my above comment:

Be sure to use `^` to only match at the beginning.

Comment (markfinger) [inline lib/map-generator.es6]: Yeah, I noticed the caret. Personally, I don't think it'd be too much of an issue as the colon after the protocol wont match any realistic OS path -even OSX tries not to support it anymore as a non-delimiter character.

That being said, the caret makes it more explicit and probably offers some minor perf improvements, so I don't mind rebasing the pull again.

**Comment by markfinger** [comment]
Ok. It's been updated with @lydell's request.

**Comment by ai** [comment]
Merged to 5.1 branch

**Pull request closed without merging**